frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx

```diff
@@ -2,148 +2,129 @@
 import { Route } from "react-router";
 
 import { renderWithProviders, screen } from "__support__/ui";
-import type { State } from "metabase-types/store";
 
+import { createMockState } from "metabase-types/store/mocks";
 import {
-  createMockSettingsState,
-  createMockState,
-} from "metabase-types/store/mocks";
-import {
+  createMockDatabase,
   createMockTokenFeatures,
   createMockUser,
 } from "metabase-types/api/mocks";
 import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
 import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import type { TokenStatusStatus } from "metabase-types/api";
 
 import { AppBanner } from "./AppBanner";
 
 interface SetupOpts {
-  shouldRenderPastDueBanner: boolean;
-  shouldRenderUnpaidBanner: boolean;
-  shouldRenderDatabasePromptBanner: boolean;
+  isAdmin: boolean;
+  tokenStatusStatus: TokenStatusStatus;
+  shouldShowDatabasePromptBanner: boolean;
 }
+const TEST_DB = createSampleDatabase();
+
+const DATA_WAREHOUSE_DB = createMockDatabase({ id: 2 });
 
 function setup({
-  shouldRenderPastDueBanner,
-  shouldRenderUnpaidBanner,
-  shouldRenderDatabasePromptBanner,
+  isAdmin,
+  tokenStatusStatus,
+  shouldShowDatabasePromptBanner,
 }: SetupOpts) {
-  renderWithProviders(<Route path="*" component={AppBanner} />, {
-    initialRoute: "/",
-    storeInitialState: createStateForConfig({
-      shouldRenderPastDueBanner,
-      shouldRenderUnpaidBanner,
-      shouldRenderDatabasePromptBanner,
-    }),
-    withRouter: true,
-  });
-}
-
-function createStateForConfig({
-  shouldRenderPastDueBanner,
-  shouldRenderUnpaidBanner,
-  shouldRenderDatabasePromptBanner,
-}: SetupOpts): State {
-  let isAdmin = false;
-  const tokenStatus = {
-    status: "Token is valid.",
-  };
-  const features = {
-    sso: false,
-  };
-
-  if (shouldRenderPastDueBanner) {
-    tokenStatus.status = "past-due";
-    isAdmin = true;
-  }
-
-  if (shouldRenderUnpaidBanner) {
-    tokenStatus.status = "unpaid";
-    isAdmin = true;
-  }
-
-  if (shouldRenderDatabasePromptBanner) {
-    isAdmin = true;
-    features.sso = true;
-    setupDatabasesEndpoints([createSampleDatabase()]);
+  if (shouldShowDatabasePromptBanner) {
+    setupDatabasesEndpoints([TEST_DB]);
+  } else {
+    setupDatabasesEndpoints([TEST_DB, DATA_WAREHOUSE_DB]);
   }
 
   const state = createMockState({
     currentUser: createMockUser({ is_superuser: isAdmin }),
-    settings: createMockSettingsState({
-      "token-features": createMockTokenFeatures(features),
-      "token-status": tokenStatus,
+    settings: mockSettings({
+      "token-status": { status: tokenStatusStatus },
+      "token-features": createMockTokenFeatures({
+        sso: shouldShowDatabasePromptBanner ? true : false,
```

Comment:
I think with the user database it shouldn't be needed? E.g. you can always return `true` or `false`. The thing is that nothing in `AppBanner` uses token features so we probably shouldn't be re-testing edge cases here once again?

Reply:
@ranquild can you clarify what you meant by
> I think with the user database it shouldn't be needed?

What do you mean by **user database**?

But I can talk about why this line is needed. This is to satisfy at least a single case to make sure the database prompt banner is rendered which is `isAdmin && isPaidPlan && !isWhiteLabeling && onlyHaveSampleDatabase` to completely make sure all cases in `AppBanner` is covered including when rendering the database prompt banner.

You can see that we need to satisfy all 4 conditions and one of that is `isPaidPlan` which we could get by either setting `sso` or `hosting` on token features to `true`
https://github.com/metabase/metabase/blob/a7fb1fdaeaa1942fb8a0ab0f672603e5de369076/frontend/src/metabase/selectors/settings.ts#L49

That's why I need to have this line. Does that make sense?